modules/core/include/opencv2/core/utils/filesystem.hpp

```diff
@@ -11,6 +11,9 @@ namespace cv { namespace utils { namespace fs {
 CV_EXPORTS bool exists(const cv::String& path);
 CV_EXPORTS bool isDirectory(const cv::String& path);
 
+CV_EXPORTS cv::String resolvePath(const cv::String& path);
+CV_EXPORTS FILE* open(const cv::String& filename, const char* mode);
```

Comment:
is this function needed ?

Reply:
It's not called yet. The basic idea was to replace `fopen` with this new function, however some decoders call `fopen` by themselves, thus we have to provide a resolved path (cf. discussion in #24491). We could remove this function. However, maybe it's worth considering to migrate the `fopen` calls inside OpenCV to this one to realize unicode support wherever needed.